src/components/hooks/useSelect.tsx

```diff
@@ -2,9 +2,50 @@ import { MenuItem } from '@blueprintjs/core';
 import { ItemRenderer } from '@blueprintjs/select';
 import { useState } from 'react';
 
-export function useSelect<T extends { label: string }>() {
-  const [value, setValue] = useState<T | null>(null);
-  const itemRenderer = getItemRenderer(value);
+type FilterType<SourceType, Type> = Pick<
+  SourceType,
+  {
+    [K in keyof SourceType]: SourceType[K] extends Type ? K : never;
+  }[keyof SourceType]
+>;
+
+interface BaseOptions<T> {
+  itemLabelKey: keyof FilterType<T, string>;
+  defaultSelectedItem?: T;
+}
+
+interface RenderOptions<T> {
+  itemLabelMap: (item: T) => string;
```

Comment:
Can you rename this to `getText`? (because it maps to the `text` prop in blueprintjs)

Reply:
Or `getItemText`